internal/slices/slices.go

```diff
@@ -0,0 +1,29 @@
+package slices
+
+import (
+	"golang.org/x/exp/constraints"
+)
+
+// Min returns the minimum value in a list.
+// It will panic if no arguments are provided.
+func Min[T constraints.Ordered](list ...T) T {
+	min := list[0]
+	for _, val := range list {
+		if val < min {
+			min = val
+		}
+	}
+	return min
+}
+
+// Chunk splits the slice into chunks of size `size`. Returns a slice of slices.
+func Chunk[T any](slice []T, size int) [][]T {
```

Comment:
If we are making changes to the interface and making everything proper, then `size` should probably be an `uint` instead:
```suggestion
func Chunk[T any](slice []T, size uint) [][]T {
```

Reply:
Theoeretically I agree, but Go actively discourages this, and you'll have to do weird type conversions. For example, you can do `make([]int, -5)` and it will compile just fine, because Go does not like specifying specific kinds of ints.

Reply:
Oh no 😢 OK, let's keep it as is.

But I still think we should probably return an error if size < 1